rules/manager.go

```diff
@@ -325,7 +331,7 @@ func (g *Group) Interval() time.Duration { return g.interval }
 // Limit returns the group's limit.
 func (g *Group) Limit() int { return g.limit }
 
-func (g *Group) run(ctx context.Context) {
+func (g *Group) run(ctx context.Context, syncForStateOverrideFunc SyncForStateOverrideFuncType) {
```

Comment:
we already have syncForStateOverrideFunc in the g, do we need to have it has a parameter?